### ISSUE TITLE
Unit tests failing under MacOSX

### ISSUE THREAD

**Issue by SkyToGround** [ess-dmsc/h5cpp#103]
The following unit tests (master branch) are currently failing under MaxOSX.

```
----------] Global test environment tear-down
[==========] 274 tests from 80 test cases ran. (130 ms total)
[  PASSED  ] 260 tests.
[  FAILED  ] 14 tests, listed below:
[  FAILED  ] Integer/0.General, where TypeParam = char
[  FAILED  ] Integer/1.General, where TypeParam = unsigned char
[  FAILED  ] Integer/2.General, where TypeParam = signed char
[  FAILED  ] Integer/3.General, where TypeParam = short
[  FAILED  ] Integer/4.General, where TypeParam = unsigned short
[  FAILED  ] Integer/5.General, where TypeParam = int
[  FAILED  ] Integer/6.General, where TypeParam = unsigned int
[  FAILED  ] Integer/7.General, where TypeParam = long
[  FAILED  ] Integer/8.General, where TypeParam = unsigned long
[  FAILED  ] Integer/9.General, where TypeParam = long long
[  FAILED  ] Integer/10.General, where TypeParam = unsigned long long
[  FAILED  ] Float/0.General, where TypeParam = float
[  FAILED  ] Float/1.General, where TypeParam = double
[  FAILED  ] Float/2.General, where TypeParam = long double

14 FAILED TESTS
```

**Comment by SkyToGround** [comment]
I think it is more likely that these failures have to do with using clang to compile the library. You are using GCC on your machines right? Jenkins is using GCC as well.

**Comment by eugenwintersberger** [comment]
Yeah I we are currently building the code with gcc and Visual Studio 2015 (at least I use the latter one on Windows).

**Comment by SkyToGround** [comment]
This appears to have been fixed in #109. I will leave this issue open until that code has been merged in the master branch.
